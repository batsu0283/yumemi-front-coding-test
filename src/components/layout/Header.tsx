type HeaderProps = {
  title: string;
};

export const Header = ({ title }: HeaderProps) => (
  <header className="bg-cyan-600 text-white p-4 mb-1 shadow-lg">
    <h1 className="text-2xl font-bold">{title}</h1>
  </header>
);
