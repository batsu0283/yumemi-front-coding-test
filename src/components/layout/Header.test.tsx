import { render, screen } from "@testing-library/react";
import { Header } from "./Header";

describe("Header", () => {
  it("指定されたタイトルが表示されること", () => {
    const testTitle = "ヘッダーのテストタイトル";
    render(<Header title={testTitle} />);
    expect(screen.getByText(testTitle)).toBeInTheDocument();
  });

  it("headerの要素が存在すること", () => {
    render(<Header title="ヘッダータイトル" />);
    expect(screen.getByRole("banner")).toBeInTheDocument();
  });

  it("Snapshot", () => {
    const { container } = render(<Header title="ヘッダータイトル" />);
    expect(container).toMatchSnapshot();
  });
});
