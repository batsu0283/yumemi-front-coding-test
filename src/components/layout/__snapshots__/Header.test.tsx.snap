// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Header > Snapshot 1`] = `
<div>
  <header
    class="bg-cyan-600 text-white p-4 mb-1 shadow-lg"
  >
    <h1
      class="text-2xl font-bold"
    >
      ヘッダータイトル
    </h1>
  </header>
</div>
`;
